cmd/coco/generate/findvaluefiles.go

```diff
@@ -3,42 +3,60 @@ package generate
 import (
 	"bytes"
 	"path/filepath"
-	"strings"
 
-	"github.com/SAP/configuration-tools-for-gitops/pkg/files"
+	"github.com/SAP/configuration-tools-for-gitops/cmd/coco/inputfile"
 	"gopkg.in/yaml.v3"
 )
 
 func readValueFiles(
-	basepath string,
+	basepath, configFileName string,
 	includeOr, includeAnd, exclude []string,
 ) (map[string]interface{}, error) {
-	fileRunner := files.New(basepath).
-		Include(files.OR, includeOr).
-		Include(files.AND, includeAnd).
-		Exclude(files.OR, exclude).
-		ReadContent()
-
-	if len(includeAnd) > 0 {
-		fileRunner = fileRunner.Include(files.AND, includeAnd)
-	}
-	vFiles, err := fileRunner.Execute()
+	valueFiles, err := inputfile.FindAll(basepath, configFileName, includeOr, includeAnd, exclude)
 	if err != nil {
 		return nil, err
 	}
-	valueFiles := vFiles.Content()
 	res := make(map[string]interface{}, len(valueFiles))
 	for path, file := range valueFiles {
 		if file.IsDir {
 			continue
 		}
-		d := yaml.NewDecoder(bytes.NewReader(file.Content))
-		var values interface{}
-		err := d.Decode(&values)
+
+		var coco inputfile.Coco
+		coco, err := inputfile.Load(path)
+		if err != nil {
+			return nil, err
+		}
+
+		if !coco.IsEnvironment() {
+			continue
+		}
+
+		dir := filepath.Dir(path)
+		var valueFilesForEnv []string
+		for _, v := range coco.Values {
+			valueFilesForEnv = append(valueFilesForEnv, dir+"/"+v+".yaml")
```

Comment:
assuming that `v` also contains the file ending `.yaml`:
```suggestion
			valueFilesForEnv = append(valueFilesForEnv, filepath.Join(dir, v))
```